paymentsheet/src/main/java/com/stripe/android/paymentsheet/DefaultFormHelper.kt

```diff
@@ -17,50 +17,102 @@ import com.stripe.android.paymentsheet.ui.transformToPaymentMethodCreateParams
+import kotlinx.coroutines.flow.MutableSharedFlow
+import kotlinx.coroutines.flow.combine
+import kotlinx.coroutines.launch
 
 internal class DefaultFormHelper(
+    private val coroutineScope: CoroutineScope,
+    private val linkInlineHandler: LinkInlineHandler,
     private val cardAccountRangeRepositoryFactory: CardAccountRangeRepository.Factory,
     private val paymentMethodMetadata: PaymentMethodMetadata,
     private val newPaymentSelectionProvider: () -> NewOrExternalPaymentSelection?,
     private val selectionUpdater: (PaymentSelection?) -> Unit,
     private val linkConfigurationCoordinator: LinkConfigurationCoordinator?,
-    private val onLinkInlineSignupStateChanged: (InlineSignupViewState) -> Unit,
 ) : FormHelper {
     companion object {
         fun create(
             viewModel: BaseSheetViewModel,
-            linkInlineHandler: LinkInlineHandler,
-            paymentMethodMetadata: PaymentMethodMetadata
+            paymentMethodMetadata: PaymentMethodMetadata,
+            linkInlineHandler: LinkInlineHandler = LinkInlineHandler.create()
         ): FormHelper {
             return DefaultFormHelper(
+                coroutineScope = viewModel.viewModelScope,
+                linkInlineHandler = linkInlineHandler,
                 cardAccountRangeRepositoryFactory = viewModel.cardAccountRangeRepositoryFactory,
                 paymentMethodMetadata = paymentMethodMetadata,
                 newPaymentSelectionProvider = {
                     viewModel.newPaymentSelection
                 },
                 linkConfigurationCoordinator = viewModel.linkHandler.linkConfigurationCoordinator,
-                onLinkInlineSignupStateChanged = linkInlineHandler::onStateUpdated,
                 selectionUpdater = {
                     viewModel.updateSelection(it)
                 },
             )
         }
 
         fun create(
+            coroutineScope: CoroutineScope,
             cardAccountRangeRepositoryFactory: CardAccountRangeRepository.Factory,
             paymentMethodMetadata: PaymentMethodMetadata,
         ): FormHelper {
             return DefaultFormHelper(
+                coroutineScope = coroutineScope,
+                linkInlineHandler = LinkInlineHandler.create(),
                 cardAccountRangeRepositoryFactory = cardAccountRangeRepositoryFactory,
                 paymentMethodMetadata = paymentMethodMetadata,
                 newPaymentSelectionProvider = { null },
                 linkConfigurationCoordinator = null,
-                onLinkInlineSignupStateChanged = {},
                 selectionUpdater = {},
             )
         }
     }
 
+    private val lastFormValues = MutableSharedFlow<Pair<FormFieldValues?, String>>()
+
+    private val paymentSelection: Flow<PaymentSelection?> = combine(
+        lastFormValues,
+        linkInlineHandler.linkInlineState,
+    ) { formValues, inlineSignupViewState ->
+        val paymentSelection = formValues.first?.transformToPaymentSelection(
```

Comment:
How do you feel about the tests for this logic? Should we extract this into a function that can be more easily unit tested?